### PR TITLE
add support for custom domains on containers

### PR DESCRIPTION
**Pull request by thomas-tacquet**

### CODE REVIEW

File: deploy/lib/createContainers.js [hunk 1]
```diff
@@ -24,6 +24,84 @@ module.exports = {
...
+        customDomains !== undefined &&
+        customDomains !== null &&
+        customDomains.length > 0
+      ) {
+        customDomains.forEach((customDomain) => {
+          domainsIdToDelete.push(customDomain.id);
+
+          let domainFound = false;
+
+          existingDomains.forEach((existingDom) => {
```
Comment (RamyChaabane) [inline deploy/lib/createContainers.js:48]: I think there is a better way to do this, for loop in a for loop is not a good practice

File: README.md [hunk 2]
```diff
@@ -344,7 +344,8 @@ You may refer to the follow examples:
 
 Custom domains allows users to use their own domains.
 
-For domain configuration please [Refer to Scaleway Documentation](https://www.scaleway.com/en/docs/compute/functions/how-to/add-a-custom-domain-name-to-a-function/)
+For domain configuration please refer to [custom domains on Function](https://www.scaleway.com/en/docs/compute/functions/how-to/add-a-custom-domain-name-to-a-function/) or
+[custom domains on Container](https://www.scaleway.com/en/docs/compute/containers/how-to/add-a-custom-domain-to-a-container/)
```
Comment (norbjd) [inline README.md:348]:
```suggestion
For domain configuration please refer to [custom domains documentation on Function](https://www.scaleway.com/en/docs/compute/functions/how-to/add-a-custom-domain-name-to-a-function/) or
[custom domains documentation on Container](https://www.scaleway.com/en/docs/compute/containers/how-to/add-a-custom-domain-to-a-container/)
```

File: deploy/lib/createContainers.js [hunk 3]
```diff
@@ -18,12 +18,81 @@ module.exports = {
...
         );
         this.waitForContainerStatus(containerIdToDelete, "deleted").then(
-          this.serverless.cli.log(`Container ${res.name} deleted`)
+          this.serverless.cli.log(`Container ${res.name} deleted`),
         );
       });
     });
   },
 
+  applyDomainsContainer(containerId, customDomains) {
```
Comment (norbjd) [inline deploy/lib/createContainers.js:27]: A test would be great :+1:

**Review by norbjd — CHANGES_REQUESTED** [review]

File: deploy/lib/createFunctions.js [hunk 4]
```diff
@@ -53,49 +54,13 @@ module.exports = {
...
-        if (
-          (customDomains === undefined || customDomains === null) &&
-          existingDomain.id !== undefined
-        ) {
-          domainsIdToDelete.push(existingDomain.id);
-        } else if (!customDomains.includes(existingDomain.hostname)) {
-          domainsIdToDelete.push(existingDomain.id);
-        }
-      });
+    this.listDomainsFunction(funcId).then((domains) => {
```
Comment (norbjd) [inline deploy/lib/createFunctions.js:60]: Is the block after `(domains) => {` the same as in the `applyDomainsContainer`. If so, can you extract the code to a function and reuse it?

**Review by norbjd — APPROVED** [review]

**Pull request merged**